### PR TITLE
Avoid having two tasks producing the same file

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

File: javaagent/build.gradle.kts [hunk 1]
```diff
@@ -218,6 +218,12 @@ tasks {
   named("generateLicenseReport").configure {
     dependsOn(cleanLicenses)
   }
+
+  // We don't publish this jar file, so the actual classifier value is not important.
+  // But we should not have two tasks (jar and shadow) that produce the same file.
+  jar {
+    archiveClassifier.set("slim")
```
Comment (iNikem) [inline javaagent/build.gradle.kts:225]: Taking suggestions for the classifier value :)

Comment (mateuszrzeszutek) [reply javaagent/build.gradle.kts:225]: `empty-do-not-use`? 😄

**Review by mateuszrzeszutek — APPROVED** [review]

File: javaagent/build.gradle.kts (hunk 1 above)
Comment (laurit) [reply javaagent/build.gradle.kts:225]: Considering https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5251 slim doesn't look like the best name for now. If this is an intermediate jar that isn't really functional then maybe `-tmp` or something else that hints that this is not supposed to be used.

Comment (iNikem) [reply javaagent/build.gradle.kts:225]: Now only to understand, why smoke test crashes...

**Comment by iNikem** [comment]
@trask I think that #5129 was a mistake:

First, `jar` is NOT needed for producing javadoc and source jars, they are totally independent.
Second, by having `jar` task with a classifier, that jar gets published and thus examples get that thin jar when they request `io.opentelemetry.javaagent:opentelemetry-javaagent`.
Third, if we have `jar` without any classifier, then things kinda work, but only by accident. See (https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4765#issuecomment-983679435).

What was the failure that #5129 tried to solve? I still think that #4765 was and is correct.

**Comment by trask** [comment]
> What was the failure that #5129 tried to solve?

maven central publishing failed during release because there was no `-javadoc` or `-sources` artifact for the `opentelemetry-javaagent` artifact, so I just reverted #4765, but I'm good with solving it a different way

**Comment by iNikem** [comment]
I will postpone this change. Don't want to mess with release process while my brain is 90% elsewhere...

**Comment by anuraaga** [comment]
I had issues with jmx-metrics project in contrib being slow so looked into this, should be superceded by #5334

**Pull request closed without merging**